### ISSUE TITLE
[N16] Lack of explicitness on data type sizes

### ISSUE THREAD

**Issue by cylon56** [compound-finance/comet#410]
The protocol heavily relies on different [sized variables](https://github.com/compound-finance/comet/blob/0f1221967149115f50a09681eea9580879ee7720/contracts/CometStorage.sol), which can have also [positive or negative values](https://github.com/compound-finance/comet/blob/0f1221967149115f50a09681eea9580879ee7720/contracts/CometCore.sol#L101) and different [scaling factors](https://github.com/compound-finance/comet/blob/0f1221967149115f50a09681eea9580879ee7720/contracts/CometCore.sol#L52). Thanks to this, the deployment and execution of the codebase will decrease gas costs.

Given the fact that it adds some complexity and undermines the readability of the codebase, it is of utter importance to maintain explicitness and consistency across different contracts.

Specifically, implicit [castings and sizes](https://github.com/compound-finance/comet/blob/0f1221967149115f50a09681eea9580879ee7720/contracts/Comet.sol#L269) should be avoided.

To improve the overall quality of the codebase, consider reviewing it in its entirety, changing all occurrences of `uint` to `uint256` and of `int` to `int256`. Consider also reviewing implicit castings from small to bigger sizes and always use the appropriate size for each variable.

**Comment by jflatow** [comment]
We removed occurrences of implicit upscaling. We accept that `uint256` and `uint` are interchangeable in Solidity and that authors are aware of this. In #454 we go even further, and also attempt to eliminate an unsafe cast which came up in the ChainSecurity audit but which we chose to keep as a gas optimization. We will leave this open until these changes have had further review (and until that PR is merged).
